src/hooks/sync-sites/use-sync-push.ts

```diff
@@ -0,0 +1,169 @@
+	} = usePullPushStates< SyncPushState >( pushStates, setPushStates );
+	const { pushStatesProgressInfo, isKeyPushing, isKeyFinished, isKeyFailed } =
+		useSyncStatesProgressInfo();
+
+	const getPushProgressInfo = useCallback(
+		async ( selectedSiteId: string, remoteSiteId: number ) => {
+			if ( ! client ) {
+				return;
+			}
+
+			const response = await client.req.get< {
+				status: 'finished' | 'failed' | 'initial_backup_started' | 'archive_import_started';
+				success: boolean;
+			} >( {
+				path: `/sites/${ remoteSiteId }/studio-app/sync/import`,
+				apiNamespace: 'wpcom/v2',
+			} );
+
+			let status: PushStateProgressInfo = pushStatesProgressInfo.importing;
+			if ( response.success && response.status === 'finished' ) {
+				status = pushStatesProgressInfo.finished;
+			} else if ( response.success && response.status === 'failed' ) {
+				status = pushStatesProgressInfo.failed;
+			}
+			// Update state in any case to keep polling push state
+			updatePushState( selectedSiteId, remoteSiteId, {
+				status,
+			} );
+		},
+		[ client, pushStatesProgressInfo, updatePushState ]
+	);
+
+	const pushSite = useCallback(
+		async ( connectedSite: SyncSite, selectedSite: SiteDetails ) => {
+			if ( ! client ) {
+				return;
+			}
+			const remoteSiteId = connectedSite.id;
+			updatePushState( selectedSite.id, remoteSiteId, {
+				remoteSiteId,
+				status: pushStatesProgressInfo.creatingBackup,
+				selectedSite,
+				isStaging: connectedSite.isStaging,
+			} );
+
+			const { archiveContent, exceedsSizeLimit } = await getIpcApi().archiveSite(
+				selectedSite.id,
+				'tar'
+			);
+			if ( exceedsSizeLimit ) {
+				getIpcApi().showErrorMessageBox( {
+					title: sprintf( __( 'Error pushing to %s' ), connectedSite.name ),
+					message: __(
+						'The site is too large to push. Please reduce the size of the site and try again.'
+					),
+				} );
+				return;
+			}
+
+			updatePushState( selectedSite.id, remoteSiteId, {
+				status: pushStatesProgressInfo.uploading,
+			} );
+
+			const file = new File( [ archiveContent ], 'loca-env-site-1.tar.gz', {
+				type: 'application/gzip',
+			} );
+			const formData = [ [ 'import', file ] ];
+			try {
+				const response = await client.req.post( {
+					path: `/sites/${ remoteSiteId }/studio-app/sync/import`,
+					apiNamespace: 'wpcom/v2',
+					formData,
+				} );
+				if ( response.success ) {
+					updatePushState( selectedSite.id, remoteSiteId, {
+						status: pushStatesProgressInfo.importing,
+					} );
+				} else {
+					console.error( response );
+					throw new Error( 'Push request failed' );
+				}
+			} catch ( error ) {
+				Sentry.captureException( error );
+				updatePushState( selectedSite.id, remoteSiteId, {
+					status: pushStatesProgressInfo.failed,
+				} );
+				getIpcApi().showErrorMessageBox( {
+					title: sprintf( __( 'Error pushing to %s' ), connectedSite.name ),
+					message: __( 'Studio was unable to connect to WordPress.com. Please try again.' ),
+				} );
```

Comment:
I'm displaying this generic error. I could also display the error that comes from the backend.

When trying to upload a bakcup bigger than the limit, we cannot grab the actual error message, because we receive a CORS error.